pkg/resource/db_instance/hooks.go

```diff
@@ -433,3 +435,179 @@ func reconcileEngineVersion(
 		a.ko.Spec.EngineVersion = b.ko.Spec.EngineVersion
 	}
 }
+
+// syncTags keeps the resource's tags in sync
+//
+// NOTE(jaypipes): RDS' Tagging APIs differ from other AWS APIs in the
+// following ways:
+//
+// 1. The names of the tagging API operations are different. Other APIs use the
```

Comment:
I'm just curious is there an api doc that can be referred ?

Reply:
> I'm just curious is there an api doc that can be referred ?

If you examine the [API reference for RDS](https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_Operations.html) you will note that RDS API has these tag-related calls:

* [AddTagsToResource](https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_AddTagsToResource.html)
* [RemoveTagsFromResource](https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_RemoveTagsFromResource.html)

whereas other APIs -- for example [EKS](https://docs.amazonaws.cn/en_us/eks/latest/APIReference/API_Operations.html) or [DynamoDB](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Operations_Amazon_DynamoDB.html) -- call these operations:

* [TagResource](https://docs.amazonaws.cn/en_us/eks/latest/APIReference/API_TagResource.html)
* [UntagResource](https://docs.amazonaws.cn/en_us/eks/latest/APIReference/API_UntagResource.html)

That's what this comment refers to.

Don't worry, though, Bruce. All the AWS APIs are inconsistent in their own little wonderful and unique ways. For example, instead of calling the list tags operation `ListTagsForResource` like almost all other APIs do, DynamoDB calls it [ListTagsOfResource](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_ListTagsOfResource.html). :shrug: 

Welcome to the wild and wonderful world of AWS APIs :)